ExampleMod/Common/Players/ExampleInventoryPlayer.cs

```diff
@@ -0,0 +1,46 @@
+﻿using ExampleMod.Content.Items;
+using System.Collections.Generic;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+using static Terraria.ModLoader.ModContent;
+
+namespace ExampleMod.Common.Players
+{
+	public class ExampleInventoryPlayer : ModPlayer
+	{
+		// AddStartingItems is a method you can use to add items to the player's starting inventory.
+		// It is also called when the player dies a mediumcore death
+		// Return an enumerable with the items you want to add to the inventory.
+		// This method adds an ExampleItem and 256 gold ore to the player's inventory.
+		// 
+		// If you know what 'yield return' is, you can also use that here, if you prefer so.
+		public override IEnumerable<Item> AddStartingItems(bool mediumCoreDeath) {
+			var testItem = new Item(ItemType<ExampleItem>());
+			var otherItem = new Item(ItemID.GoldOre) {
+				stack = 256
+			};
+
+			if (mediumCoreDeath) {
+				return new[] {
+					new Item(ItemID.HealingPotion)
+				};
+			}
+
+			return new[] {
+				testItem,
+				otherItem
+			};
```

Comment:
Please combine unnecessary variables, especially to declare them and then return without using them. Just `return new[] {new Item(...), new Item(...) {stack = 256 }}`